GH-89455: Add missing attributes (added in 3.11) to traceback module docs (#105044)
python/cpython: Doc/library/traceback.rst

   Note that when locals are captured, they are also shown in the traceback.

   *max_group_width* and *max_group_depth* control the formatting of exception

>    groups. The depth refers to the nesting level of the group, and the width

Reviewer: ```suggestion
   groups (see :exc:`BaseExceptionGroup`). The depth refers to the nesting level of the group, and the width
```

>    groups (see :exc:`BaseExceptionGroup`). The depth refers to the nesting level of the group, and the width

Reviewer: Wrap to 80 chars (PEP8)

      A :class:`TracebackException` of the original ``__context__``.

   .. attribute:: exceptions

>       For exception groups - a list of :class:`TracebackException` of the original ``exceptions``.

Reviewer: ```suggestion
      If ``self`` represents an :exc:`ExceptionGroup` - a list of :class:`TracebackException` instances representing the nested exceptions.
```

>       If ``self`` represents an :exc:`ExceptionGroup` - a list of :class:`TracebackException` instances representing the nested exceptions.

Reviewer: Wrap to 80 chars (PEP8)

      A :class:`TracebackException` of the original ``__context__``.

   .. attribute:: exceptions

      If ``self`` represents an :exc:`ExceptionGroup` - a list of

>       :class:`TracebackException` instances representing the nested exceptions.

Reviewer: ```suggestion
      If ``self`` represents an :exc:`ExceptionGroup`, this field holds a list of
      :class:`TracebackException` instances representing the nested exceptions.
      Otherwise it is ``None``.
```

      For syntax errors - the offset into the text where the error occurred.

   .. attribute:: end_offset

      For syntax errors - the end offset into the text where the error occurred.
      Can be ``None`` if not present.

>       .. versionadded:: 3.10

Reviewer: It might be worth putting the 3.10 changes in a separate PR that can be backported all the way back to 3.10.

Author: Moved 3.10 changes to #105046
